uwblueprint/paramedics-react
# Add CCP Dashboard Screen 2

## Overview
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

[Notion task 1](https://www.notion.so/uwblueprintexecs/CCP-Dashboard-Screen-2-f600fbc4a87442c4a23684da964823cd)
[Notion task 2](https://www.notion.so/uwblueprintexecs/transportedAt-field-for-patients-4bcbc2ca9459478f8093e14b241b69dd)

## Changes
- Add hospital filtering and sorting on CCP Dashboard Screen 1 (Patient Overview tab)
- Add CCP Dashboard Screen 2 (Hospital tab)
- Refactor components to be reusable between `PatientOverview` and `HospitalOverview`
- Move loading state outside of tabs

**Note:** The app bar in the design has not been implemented (it's part of a separate PR)

## Testing
- Make sure you've pulled the latest from master of `paramedics-web` and the container is up
- Add some patients to a specific CCP with varying triage levels, statuses, and hospitals
- Navigate to /events/:eventId/ccps/:ccpId where eventId and ccpId are valid (ccpId must be associated with the eventId)
- See the loading state when patients are not loaded

On the "Patient Overview" tab:
- Try sorting by hospital
    - **Note**: null values will always be last when sorting by column
- Try filtering by hospital
    - The filter "popover" should show all unique hospital names in alphabetical order
    - Selected filters (via checkboxes) are not applied until user clicks "Apply Filter" button
    - When no checkboxes are selected for a particular category, it means the patients are not filtered by that category
    - Chips appear that match the selected AND applied filters
    - Clicking "X" on a chip will remove that filter (also check that the state of the checkboxes in the popover will reflect this)

On the "Hospital" tab:
- Should only see hospitals with "Transported" status
- See default in dropdown is "All Hospitals"
- The dropdown menu should include all unique hospital names in alphabetical order
- Select different options from the dropdown menu and verify the following
    - Only patients with the selected hospital value are shown
    - Number of (filtered) patients is reflected correctly
    - CCP Triage Card reflects the patients that belong to the selected hospital only 

## Screenshots
Loading state:
![image](https://user-images.githubusercontent.com/22457859/87752720-0d172600-c7cf-11ea-8df3-a78123a4bb12.png)

Loaded CCP Dashboard Screen 1:
![image](https://user-images.githubusercontent.com/22457859/87752774-2d46e500-c7cf-11ea-96d6-65a5dfc68a72.png)
![image](https://user-images.githubusercontent.com/22457859/87752794-36d04d00-c7cf-11ea-9836-158b3dff31a8.png)
![image](https://user-images.githubusercontent.com/22457859/87753003-9fb7c500-c7cf-11ea-85eb-09804265d916.png)

Loaded CCP Dashboard Screen 2:
![image](https://user-images.githubusercontent.com/22457859/87795200-b4b74700-c815-11ea-8040-b067ccd2338a.png)

UPDATED CCP Dashboard Screen 2 (table columns):
![image](https://user-images.githubusercontent.com/22457859/88488166-8e9e4f00-cf59-11ea-96da-416e91b5dee6.png)
